cli/tests/fixtures/extensions/api-extension/phylum-api.ts

```diff
@@ -0,0 +1,36 @@
+// This helper file wraps Deno.core.opAsync calls for ergonomics. There is no
+// concept of injecting a module in Deno, so clients won't be able to use something
+// like `import * as PhylumApi from 'phylum-api'` and will have to manually include
+// this file instead.
```

Comment:
I tried to do that in a few potential variations, but in the end a valid URL was required every time -- iiuc Deno has no concept that would allow `import 'phylum-api'` directly, it has to be an existing and resolvable URL. The [FAQ](https://deno.land/manual/linking_to_external_code#faq) seems to also corroborate this, while not stating it explicitly. I tried specifying e.g. `./phylum-api.ts` and then adding a code module to the runtime via `include_str!` but Deno errored mentioning that the file did not exist. I guess we can dig deeper on this, e.g. how is the top-level scope filled (the one that contains `Deno`)? Maybe we could check if we can evaluate some JS code that fills that scope?

Reply:
> in the end a valid URL was required every time

You need a URI for modules to be imported, that is correct. But that URI doesn't actually need to be a valid file path. I tried that out when I was messing around with the WASM stuff. The path itself is resolved and read by the module loader, which is provided by us, so we should be able to inject from there.

Reply:
Oh, I had not understood that. So I guess we can make a special case in the module loader which returns our shim code if the module is e.g. `phylum-api`. I'll try that.

Reply:
> Oh, I had not understood that. So I guess we can make a special case in the module loader which returns our shim code if the module is e.g. phylum-api. I'll try that.

Yeah that's actually exactly what I was thinking, but I haven't tried it. All I tried was passing a fake module URI to the module loader when loading just a single module with the default loader just directly from a string. So it should be possible to have a module loader that accepts non-paths.